eox_core/api/v1/permissions.py

```diff
@@ -40,7 +40,11 @@ def has_permission(self, request, view):
         To grant access, checks if the requesting user either can call eox-core API or if it's an admin user.
         """
         try:
-            allowed_for_site = request.user.is_staff or request.get_host() in request.auth.client.url
+            if request.user.is_staff:
+                allowed_for_site = request.user.is_staff
```

Comment:
This first part of the if is just confusing. If the is in fact staff we are making the whole function take a different path. My suggestion would be to completely remove from the logic by looking that up first and returning.
If the user is not staff, then you can do the auth.application or auth.client checks each in a try..catch block